packages/server/src/logger/types.ts

```diff
@@ -21,32 +22,63 @@ export class Logger {
     [LoggerLevel.Info]: clc.green
   }
 
-  constructor(private scope: string) {}
+  constructor(private scope: string, private logPrefix?: string) {}
 
   sub(scope: string) {
     return new Logger(`${this.scope}:${scope}`)
   }
 
+  prefix(prefix: string) {
+    return new Logger(this.scope, prefix)
+  }
+
   info(message: string, data?: Param) {
-    this.print([message, data], LoggerLevel.Info)
+    this.printPrefix([message, data], LoggerLevel.Info)
   }
 
   debug(message: string, data?: Param) {
-    this.print([message, data], LoggerLevel.Debug)
+    this.printPrefix([message, data], LoggerLevel.Debug)
   }
 
   warn(message: string, data?: Param) {
-    this.print([message, data], LoggerLevel.Warn)
+    this.printPrefix([message, data], LoggerLevel.Warn)
+  }
+
+  window(lines: string[], level = LoggerLevel.Info) {
+    const line = '========================================'
+    this.print(
+      [
+        line +
+          '\n' +
+          lines
+            .map((x) => this.center(this.logPrefix ? `${clc.blackBright(this.logPrefix)}${x}` : x, line.length))
+            .join('\n') +
+          '\n' +
+          this.center(line, line.length)
+      ],
+      level
+    )
   }
 
-  error(error: undefined, message: string, data?: Param): void
-  error(error: Error, message?: string, data?: Param): void
   error(error: Error | undefined, message?: string, data?: Param) {
```

Comment:
Because it's not used anywhere. The overloading doesn't seem to do anything either. What different behavior can we get from the overloads? It seems like you can call the original function in a the same ways as the overloads.

Reply:
The difference is that if you provide undefined as the error parameter, you are forced to specify a message, which is more user-friendly and fits better with how the function handles its parameters.

Reply:
Makes sense but it's not really something that is likely going to happen. Another alternative could be to make the message required as it is with every other log level.